core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -302,6 +305,98 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   }
 
   /**
+   * testSendBeforeAndAfterPartitionExpansion checks the partitioning behavior before and after partitions are added
```

Comment:
Nit: no need to have the test name here, it doesn't make the comment clearer and it's one more thing that can go out of date if we rename the test.